addons/website_sale/models/website_snippet_filter.py

```diff
@@ -111,7 +111,12 @@ def _get_products_latest_viewed(self, website, limit, domain, context):
                     domain,
                     [('id', 'in', products_ids)],
                 ])
-                products = self.env['product.product'].with_context(display_default_code=False, add2cart_rerender=True).search(domain, limit=limit)
+                # `search` will not keep the order of tracked product => re-sort at the end
+                order = dict(zip(reversed(products_ids), range(len(products_ids), 0, -1)))
+                products = self.env['product.product'].with_context(
+                    display_default_code=False,
+                    add2cart_rerender=True,
+                ).search(domain, limit=limit).sorted(lambda p: order[p.id])
```

Comment:
```suggestion
                products = self.env['product.product'].with_context(
                    display_default_code=False,
                    add2cart_rerender=True,
                )._search(domain, limit=limit)
                # `search` will not keep the order of tracked products => keep the order
                products = self.env['product.product'].browse(
                    [product_id for product_id in product_ids if product_id in products]
                )
```
something like that might be easier to read and understand IMHO

Reply:
wait

Reply:
here you go 😂